src/workspaces/features/RunPipelineDialog/RunPipelineDialog.tsx

```diff
@@ -49,31 +76,32 @@ const RunPipelineDialog = (props: RunPipelineDialogProps) => {
       clearCache();
       onClose();
     },
-    getInitialState() {
-      const version =
-        ("version" in props && props.version) ||
-        ("run" in props && props.run.version) ||
-        pipeline.currentVersion ||
-        null;
-
-      let state: any = {
-        version,
-      };
-
-      if ("run" in props && props.run) {
-        state = { ...state, ...props.run.config };
-      }
-
-      return state;
-    },
   });
   const { t } = useTranslation();
 
   useEffect(() => {
     if (!open) {
       form.resetForm();
+      setShowVersionPicker(false);
+    } else if (!("version" in props)) {
+      fetch({
+        variables: {
+          workspaceSlug: pipeline.workspace?.slug,
+          pipelineCode: pipeline.code,
+        },
+      });
     }
-  }, [open, form]);
+  }, [open, form, fetch, props, pipeline]);
+
+  useEffect(() => {
+    if ("version" in props && props.version) {
+      form.setFieldValue("version", props.version);
+    } else if ("run" in props && props.run) {
+      form.setFieldValue("version", props.run.version);
+    } else {
+      form.setFieldValue("version", data?.pipelineByCode?.currentVersion);
+    }
+  }, [form, data, props]);
```

Comment:
It means it would be triggered every time the props are changed. Not a good idea :/

Reply:
Yes, you're right, can trigger a lot of update